appengine/java/com/google/auth/appengine/AppEngineCredentials.java

```diff
@@ -51,7 +52,7 @@
  *
  * <p>Fetches access tokens from the App Identity service.
  */
-public class AppEngineCredentials extends GoogleCredentials {
+public class AppEngineCredentials extends GoogleCredentials implements ServiceAccountSigner {
```

Comment:
While not all credential types can sign, is it worth considering adding the signing methods and a boolean to check whether it is supported to the high level Credentials base type? If a credential can do it, the behavios is strongly tied to the identity.That might simplify the hierarchy, although the interface is certainly more clean and targeted. Interested in your thoughts on the pros/cons here.